firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -283,12 +286,15 @@ extension BrowserViewController: WKUIDelegate {
                        currentTab: Tab,
                        webView: WKWebView) -> [UIAction] {
         let actionBuilder = ActionProviderBuilder()
+        let isJavascriptScheme = (url.scheme?.caseInsensitiveCompare("javascript") == .orderedSame)
 
-        if !isPrivate {
+        if !isPrivate && !isJavascriptScheme {
             actionBuilder.addOpenInNewTab(url: url, currentTab: currentTab, addTab: addTab)
         }
 
-        actionBuilder.addOpenInNewPrivateTab(url: url, currentTab: currentTab, addTab: addTab)
+        if !isJavascriptScheme {
+            actionBuilder.addOpenInNewPrivateTab(url: url, currentTab: currentTab, addTab: addTab)
+        }
```

Comment:
Reviewer note: this is the only logic change in this ticket, most of the other diff is just refactors to try to break apart our contextual menu code which was getting fairly large. (cc @issammani)